File: src/main/java/org/cloudfoundry/credhub/constants/CredentialWriteMode.java
package org.cloudfoundry.credhub.constants;

import com.fasterxml.jackson.annotation.JsonValue;

public enum CredentialWriteMode {
  OVERWRITE("overwrite"),
  NO_OVERWRITE("no-overwrite"),
  CONVERGE("converge");

  public final String mode;

  CredentialWriteMode(String mode) {
    this.mode = mode;
  }

  @JsonValue
  public String forJackson() {
    return mode;
  }

  @Override
  public String toString() {
    return mode;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/constants/CredentialWriteMode.kt
package org.cloudfoundry.credhub.constants

import com.fasterxml.jackson.annotation.JsonValue

enum class CredentialWriteMode private constructor(
    val mode: String,
) {
    OVERWRITE("overwrite"),
    NO_OVERWRITE("no-overwrite"),
    CONVERGE("converge"),
    ;

    @JsonValue
    fun forJackson(): String = mode

    override fun toString(): String = mode
}
